src/main/java/hudson/plugins/favorite/Favorites.java

```diff
@@ -0,0 +1,131 @@
+package hudson.plugins.favorite;
+
+import com.google.inject.Singleton;
+import hudson.model.Item;
+import hudson.model.User;
+import hudson.plugins.favorite.user.FavoriteUserProperty;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+
+/**
+ * Public API for Favorites
+ */
+public final class Favorites {
```

Comment:
@michaelneale overnight I wrote a nicer public API for Favorites that hides all the rubbish you need todo with the user property.